dotCMS/src/integration-test/java/com/dotmarketing/startup/runonce/Task230523CreateVariantFieldInContentletIntegrationTest.java

```diff
@@ -0,0 +1,120 @@
+                existsVariantId = true;
+            }
+        }
+
+        assertTrue(existsVariantId);
+    }
+
+    @Test
+    public void executeTaskUpgrade() throws SQLException, DotDataException, IOException {
+        final DotConnect dotConnect = new DotConnect();
+
+        final ContentType contentType = new ContentTypeDataGen().nextPersisted();
+        final Contentlet contentlet_1 = new ContentletDataGen(contentType).nextPersisted();
+        final Contentlet contentlet_2 = new ContentletDataGen(contentType).nextPersisted();
+
+        final String asJson = APILocator.getContentletJsonAPI().toJson(contentlet_1);
+        final Map<String, Object> contentletAsMap = JsonUtil.getJsonFromString(asJson);
+        contentletAsMap.put(Contentlet.VARIANT_ID, "1");
+
+        dotConnect.setSQL("UPDATE contentlet SET contentlet_as_json = ? WHERE inode = ?")
+                .addJSONParam(contentletAsMap)
+                .addParam(contentlet_1.getInode())
+                .loadResults();
+
+        dotConnect.setSQL("UPDATE contentlet SET contentlet_as_json = NULL WHERE inode = ?")
+                .addParam(contentlet_2.getInode())
+                .loadResults();
+
+        final Task230523CreateVariantFieldInContentlet task230523CreateVariantFieldInContentlet =
+                new Task230523CreateVariantFieldInContentlet();
+
+        assertFalse(task230523CreateVariantFieldInContentlet.forceRun());
+
+
+        dotConnect.executeStatement("ALTER TABLE contentlet DROP COLUMN variant_id");
+
+        assertTrue(task230523CreateVariantFieldInContentlet.forceRun());
+
+        task230523CreateVariantFieldInContentlet.executeUpgrade();
+
+        assertFalse(task230523CreateVariantFieldInContentlet.forceRun());
+
+        final DotDatabaseMetaData databaseMetaData = new DotDatabaseMetaData();
+        assertTrue(databaseMetaData.hasColumn("contentlet", "variant_id"));
+
+        final ArrayList<Map<String, Object>> contentlets_1 = dotConnect.setSQL(
+                        "SELECT contentlet_as_json FROM contentlet WHERE inode = ?")
+                .addParam(contentlet_1.getInode())
+                .loadResults();
+
+        assertEquals(1, contentlets_1.size());
+
+        final Map<String, Object> contentletAsMapFromDB = JsonUtil.getJsonFromString(
+                contentlets_1.get(0).get("contentlet_as_json").toString());
+        assertNull(contentletAsMapFromDB.get(Contentlet.VARIANT_ID));
+
+
+        final ArrayList<Map<String, Object>> contentlets_2 = dotConnect.setSQL(
+                        "SELECT contentlet_as_json FROM contentlet WHERE inode = ?")
+                .addParam(contentlet_2.getInode())
+                .loadResults();
+
+        assertEquals(1, contentlets_2.size());
+
+        final Object contentletAsJson = contentlets_2.get(0).get("contentlet_as_json");
+        assertFalse(contentletAsJson != null && UtilMethods.isSet(contentletAsJson.toString()));
+    }
```

Comment:
I think an assert for the actual value of the `variant_id` column is missing

Reply:
done https://github.com/dotCMS/core/pull/25035/commits/4a7b5555a6623425addbb8ea468a2a3201b4d144#diff-cf4d6acc2cfbfbdb63e920244db08e5d1193baaf541f4f9eb3ecedbc3b75147eR107-R124